add jalbum 37.7

silmaril42 opened the pull request:
## Things done

Add new package `jalbum`.

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and others READMEs.

silmaril42 committed:
jalbum: init at 37.7

silmaril42 commented:
I think the checks are failing, because I added the maintainer `silmaril` in another PR #426712, which has not been merged yet.
I hope this will resolve itself with merging that PR.

Sorry if this is causing additional effort! I am still trying to find my way around this project.

thtrf commented:
Don't create a separate PR to add yourself as a maintainer. You can simply do this in the first commit of this PR: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md#how-to-become-a-maintainer

silmaril42 committed:
maintainers: add silmaril

silmaril42 commented:
> Don't create a separate PR to add yourself as a maintainer. You can simply do this in the first commit of this PR: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md#how-to-become-a-maintainer

I misread the documentation and created a separate PR instead of a separate commit.

Of course, it makes more sense to have everything in one PR.

silmaril42 commented:
Sorry, I think I am lacking the git knowledge on how to rearrange the existing commits.
My only idea of how to clean this up would be creating a new PR.

About the commit you'd like me to remove: 
My original change to `maintainer-list.nix` led to a failing test because of formatting issues.
The commit was the result of running `nix-shell --run treefmt`.
I can see that we probably don't want to have all those whitespace changes in the PR.
How can I make sure the formatting is accepted by the test without reformatting the whole file?

thtrf commented:
You don't need to create a new PR, you can just use `git rebase`: https://git-scm.com/docs/git-rebase#_interactive_mode

In your original change to `maintainer-list.nix`, you removed whitespaces that were present before. You should just add yourself to the list.

silmaril42 commented:
Interesting... I never used interactive rebase before, but it looks like a very powerful tool, if you understand how to use it.

Thanks for the learning opportunity :smile: 

I hope the code has reached an acceptable state now?

thtrf commented:
You have still changed parts of the comment in `maintainer-list.nix`. Please remove these changes.

thtrf commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 426724`
Commit: `8078e4b9e81e86bbd39d0a01ce6b88b05521e798`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>jalbum</li>
  </ul>
</details>

thtrf commented on pkgs/by-name/ja/jalbum/package.nix:
```diff
@@ -0,0 +1,84 @@
+
+  src = fetchzip {
+    url = "https://download.jalbum.net/download/${version}/jAlbum.zip";
+    sha256 = "QGy8agZ88Q0npcodnP6S/BF4lQEGbUMlQa8v7T2kNWU=";
```
There is a hash mismatch.

thtrf commented on pkgs/by-name/ja/jalbum/package.nix:
```diff
@@ -0,0 +1,84 @@
+  common-updater-scripts,
+  curl,
+  gnused,
+}:
```
```suggestion
}:

```

thtrf commented:
Apart from my suggestions, LGTM.

silmaril42 committed:
update hash for JAlbum 37.7.3

This might be a problem: Under the same download URL for version 37.7 new versions with a change in the third digit are provided. I can see no way to determine this version number while downloading.

Co-authored-by: thtrf <82712122+thtrf@users.noreply.github.com>

silmaril42 committed:
pkgs/by-name/ja/jalbum/package.nix

Co-authored-by: thtrf <82712122+thtrf@users.noreply.github.com>

silmaril42 committed:
Update pkgs/by-name/ja/jalbum/package.nix

Co-authored-by: thtrf <82712122+thtrf@users.noreply.github.com>

silmaril42 commented:
I am not sure how we can handle this:
The download version changes without a change in the version number on the website.
They only tell us it's 37.7, while the content is updated and changes the third digit (currently 37.7.3), which we see in the splash screen of the starting application (I don't know yet, if there are other places with this information).

thtrf commented:
Oh, that's a major problem

silmaril42 commented:
I got a first answer: 
There is this URL: https://jalbum.net/pad.xml
The XML contains only the latest release of the first two version digits, which isn't much more helpful than what we get by scraping the headline of the download page.

My question was forwarded to a lead developer who will hopefully get back to me with a better solution.
This might take some days because of vacation times.

So, let's be patient and see what we'll get.

silmaril42 commented:
Update: I exchanged some mails with the JAlbum people. They seems genuinely interested in improving the installation and upgrade experience for all kinds of Linux.

It might be some time before we get something tangible - I'll get back to this PR when I think it makes sense to try some new possibility.
